bitcoin/src/psbt/raw.rs

```diff
@@ -162,7 +162,11 @@ where
     fn consensus_decode<R: Read + ?Sized>(r: &mut R) -> Result<Self, encode::Error> {
         let prefix = Vec::<u8>::consensus_decode(r)?;
         let subtype = Subtype::from(r.read_u8()?);
-        let key = read_to_end(r)?;
+
+        // The limit is a DOS protection mechanism the exact value is not
+        // important, 1024 bytes is bigger than any key should be.
```

Comment:
Makes sense. And we should impose a limit on all PSBT vecs. Further, I think some sane limits should be documented in a BIP.

Reply:
I'm just doing the minimum changes possible to `psbt` because dev has moved from this repo.

Reply:
Yes, just some thinking aloud.